testing/e2e.sh

```diff
@@ -62,6 +62,77 @@ EOF
   bazel build --verbose_failures --spawn_strategy=standalone :pause_based
 }
 
+# We test this out-of-line because of the nonsense requiring go_prefix
+# to be defined in //:go_prefix.  This means we can't test Go in a repo
+# defining repository rules without requiring all downstream consumers
+# to import Go unnecessarily.
+function test_go_image() {
+  local directory=$(mktemp -d)
+
+  cd "${directory}"
+
+  cat > "WORKSPACE" <<EOF
+workspace(name = "go_image")
+
+local_repository(
+    name = "io_bazel_rules_docker",
+    path = "$ROOT",
+)
+load(
+  "@io_bazel_rules_docker//docker:docker.bzl",
+  "docker_repositories",
+)
+docker_repositories()
+
+# We must load these before the go_image rule.
+git_repository(
+    name = "io_bazel_rules_go",
+    remote = "https://github.com/bazelbuild/rules_go.git",
+    tag = "0.4.4",
```

Comment:
I know :( unfortunately it is not on our roadmap yet. I think we should build some kind of tools that would update those versions callable from bazel e.g. `bazel tools bump-dep-versions` o